libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -1037,6 +1049,31 @@ int SocketAPM_accept(lua_State *L) {
     return 0;
 }
 
+/*
+  convert a uint32_t ipv4 address to a string
+ */
+int SocketAPM_ipv4_addr_to_string(lua_State *L) {
+    binding_argcheck(L, 1);
+    const uint32_t ip_addr = get_uint32(L, 1, 0, UINT32_MAX);
+    static char buf[IP4_STR_LEN];
```

Comment:
```suggestion
   char buf[IP4_STR_LEN];
```
`IP4_STR_LEN` is only 16 and `static` is a footgun here IMO.

Reply:
you're right, it was cut and paste from another file, fixed